### PR TITLE
Reduce trival pull flooding by restricting requirement

### PR DESCRIPTION
**Pull request by yue9944882**
if you notice, more & more trival pulls is flooding for requesting a membership which is sometimes costy. add this restriction might possibly help frequency of trival pulls.

/cc @nikhita

### CODE REVIEW

**Comment by cblecker** [comment]
@yue9944882 Is this something that we're actively seeing? This is normally something that the sponsors would deal with, as the sponsors should have close interactions with the person they are sponsoring.

**Comment by yue9944882** [comment]
@cblecker yep. it's flooding since the beginning of this year. mostly typo fix, updating err msgs, etc.

[https://github.com/kubernetes/kubernetes/pull/59464](https://github.com/kubernetes/kubernetes/pull/59464)

at the bottom^^, a lot of typofixes was conflicting which are closed by @nikhita manually one by one. 🙃

File: community-membership.md [hunk 1]
```diff
@@ -46,7 +46,7 @@ remain active contributors to the community.
       forums, Stack Overflow)
 - Subscribed to [kubernetes-dev@googlegroups.com]
 - Have read the [contributor guide]
-- Actively contributing to 1 or more subprojects.
+- Actively contributing non-trival pulls to 1 or more subprojects.
```
Comment (nikhita) [inline community-membership.md]: I'm concerned that non-trivial is subjective. There is no strict codified definition of what is non-trivial and what isn't. While we can agree that typos are trivial, the line soon blurs down.

Like @cblecker  mentioned, I would also expect that if an existing member is sponsoring someone to become a member, they would only do so if they have had close interactions with the non-member and they think that the non-member's contributions are "non-trival enough according to them" for warranting membership.

**Comment by nikhita** [comment]
> at the bottom^^, a lot of typofixes was conflicting which are closed by @nikhita manually one by one.

https://github.com/kubernetes/kubernetes/pull/59464 fixed _all_ typos in the project (which were present at that time) and added a script to catch them. I went ahead and closed down all PRs which were created to fix typos (just a cleanup). Some of these were created almost a year ago. Given that all of these were created over time, I think the number of typo fixes are not _that_ overwhelming.

> members keep opening trival pulls are encouraging/lgtming others to do follow the ways.

Drive by contributions are a common part of OSS. I think we shouldn't completely discourage "trivial contributions". Contributors don't always start out serious, they start out curious. :)

@yue9944882 do you see a real concern where members have created _only_ typo fixes? (also, just a quick note: even if that's the case, please don't call them out publicly. You can reach out to @cblecker or me privately on slack :))

**Comment by yue9944882** [comment]
> Contributors don't always start out serious, they start out curious. :)

yeah. starting with curiosity is always good and these pulls are always welcome. but if one keeps curious for months, it'll be another problem. as a rule of thumb, we usually ignore trival pulls to prevent it from flooding in some other open-source projects. but in k/k we are granting `lgtm` privilege to them, this leads to more trival pulls and an uncertainty in the code quality. 

imho contributors are expected to have fundamental coding skill & a *basic* knowledge of codebase before getting start and opening pulls. but raising trival pulls for months is obviously a correct direction to go.

> even if that's the case, please don't call them out publicly.

of course 😅

**Comment by yue9944882** [comment]
in a word, to control the quality of membership bandwidth

File: community-membership.md (hunk 1 above)
Comment (yue9944882) [reply community-membership.md]: > I'm concerned that non-trivial is subjective.

indeed 😅at least we can start by typofix

**Comment by castrojo** [comment]
We have a guideline on how to handle trivial edits: https://github.com/anothertobi/community/blob/master/contributors/guide/pull-requests.md#10-trivial-edits

I think it's just a matter of finding a clever way to point it out to people.

**Comment by yue9944882** [comment]
this trival pulls flooding is now likely driven by [contribution ranking](https://k8s.devstats.cncf.io/d/12/dashboards?refresh=15m&orgId=1). especially in china, companies are using that ranking to bid for projects for profits. hopefully we could clarify that ranking w/ some more details instead of simply counting amount of commits.

**Comment by cblecker** [comment]
I’d love to see some data on this, so we can track how much of an issue this is.

Can we open an issue on this so we can track/discuss, and then decide on the right course of action?

**Comment by nikhita** [comment]
https://github.com/kubernetes/community/issues/2953 is the tracking issue on this. I think the general consensus is that we need data before taking any concrete steps on this front.

Closing this issue until we have more data on it. Please feel free to reopen if needed.
/closw

**Pull request closed without merging**